admin/create-dumps.sh

```diff
@@ -32,12 +41,26 @@ function add_rsync_include_rule {
     echo "$SHA256_FILE_RULE" >> "$RULE_FILE"
 }
 
-echo "This script is being run by the following user: "; whoami
+# all cleanup code should be placed within this function
+function on_exit {
+    echo "Disk space when create-dumps ends:"; df -m
+
+    if [ -n "$TMPDIR" ]; then
+        rm -rf "$TMPDIR"
```

Comment:
@atj is this dangerous if someone sets TMPDIR to `/` ? (in our containers we run things as root). Might be good to have some more validation here

Reply:
Good spot. Yes this is dangerous because there is no guarantee that `TMPDIR` was set by the script before `on_exit` is called. Unfortunately the mitigation added here can be trivially overcome because it doesn't take relative paths into account (e.g. `/../`).

My suggestion would be to explicitly set TMPDIR to the empty string at the top of the script (with an explanatory comment), so that even if an attacker has injected it into the environment it is guaranteed to be set to a safe value whenever `on_exit` is called.